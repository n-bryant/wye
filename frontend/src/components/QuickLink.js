import React from "react";
import PropTypes from "prop-types";

import createClassNameHelper from "@n_bryant/classnames-helper";
import JSS_CLASS_NAME_PREFIX from "../../lib/classNamePrefix";
import { withStyles } from "@material-ui/core/styles";

import Link from "next/link";
import Button from "@material-ui/core/Button";
import Typography from "@material-ui/core/Typography";

import styles from "./QuickLink.styles";

/**
 * renders a Button wrapped with a Link that serves as a Quick Link
 */
export const QuickLink = props => {
  const classnames = QuickLink.classnames(props);
  const { label, linkHref, linkAs } = props;

  return (
    <React.Fragment>
      {linkAs ? (
        <Link href={linkHref} as={linkAs}>
          <a className={classnames.root()}>
            <Button className={classnames.element("button")}>
              <Typography variant="body1">{label}</Typography>
            </Button>
          </a>
        </Link>
      ) : (
        <Link href={linkHref}>
          <a className={classnames.root()}>
            <Button className={classnames.element("button")}>
              <Typography variant="body1">{label}</Typography>
            </Button>
          </a>
        </Link>
      )}
    </React.Fragment>
  );
};
QuickLink.classnames = createClassNameHelper(
  `${JSS_CLASS_NAME_PREFIX}QuickLink`
);
QuickLink.propTypes = {
  // styles to apply
  classes: PropTypes.shape({
    root: PropTypes.string,
    button: PropTypes.string
  }),
  // the label to display on the Button
  label: PropTypes.string.isRequired,
  // the href value for the Link
  linkHref: PropTypes.string.isRequired,
  // the as value for the Link
  linkAs: PropTypes.string
};
QuickLink.defaultProps = {
  classes: {}
};

// apply styles
export const StyledQuickLink = withStyles(styles)(QuickLink);

export default StyledQuickLink;
