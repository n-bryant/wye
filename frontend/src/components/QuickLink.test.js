import Link from "next/link";
import { QuickLink } from "./QuickLink";

describe("QuickLink", () => {
  const props = {
    classes: {
      root: "root",
      button: "button"
    },
    label: "label",
    linkHref: "linkHref"
  };

  it("should render successfully", () => {
    const wrapper = shallow(<QuickLink {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should render a Link with an as prop value matching the provided linkAs, if linkAs is defined", () => {
    const propsWithLinkAs = {
      ...props,
      linkAs: "linkAs"
    };
    const wrapper = shallow(<QuickLink {...propsWithLinkAs} />);
    expect(wrapper.find(Link).prop("as")).toBe(propsWithLinkAs.linkAs);
  });
});
