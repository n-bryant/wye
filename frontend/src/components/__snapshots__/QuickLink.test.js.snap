// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuickLink should render successfully 1`] = `
<Fragment>
  <Link
    href="linkHref"
  >
    <a
      className="wye-QuickLink root"
    >
      <WithStyles(ForwardRef(Button))
        className="wye-QuickLink__button button"
      >
        <WithStyles(ForwardRef(Typography))
          variant="body1"
        >
          label
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Button))>
    </a>
  </Link>
</Fragment>
`;
